spec/system/page_objects/admin_user_badges.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module PageObjects
+  module Pages
+    class AdminUserBadges < PageObjects::Pages::Base
+      def visit_page(user)
+        page.visit "/admin/users/#{user.id}/#{user.username}/badges"
+        self
+      end
+
+      def find_badge_row_by_granter(granter)
+        page.find(:table_row, { "Granted By" => "#{granter.username}" })
```

Comment:
This is interesting, I didn't know there was a way to find a `table_row`! Only suggestion I have here is to scope it to the table's top level CSS class e.g. something like `page.find(".badges-granted").find....`